File: android/robot/src/main/java/org/openbot/env/ILocalConnection.java
package org.openbot.env;

import android.content.Context;

public interface ILocalConnection {

  void init(Context context);

  void setDataCallback(IDataReceived dataCallback);

  void connect(Context context);

  void disconnect(Context context);

  boolean isConnected();

  void sendMessage(String message);

  void stop();

  void start();

  boolean isVideoCapable();
}

File: android/controller/src/main/java/org/openbot/controller/ILocalConnection.kt
package org.openbot.controller

import android.content.Context

interface ILocalConnection {

    fun init (context: Context)

    fun setDataCallback(dataCallback: IDataReceived?)

    fun connect(context: Context)

    fun disconnect(context: Context? = null)

    fun isConnected(): Boolean

    fun sendMessage(message: String?)
}